src/sfProject.ts

```diff
@@ -123,14 +123,18 @@ export class SfProjectJson extends ConfigFile {
   }
 
   public async write(newContents?: ConfigContents): Promise<ConfigContents> {
-    this.setContents(newContents);
+    if (newContents) {
+      this.setContents(newContents);
+    }
     this.validateKeys();
     await this.schemaValidate();
     return super.write();
   }
 
   public writeSync(newContents?: ConfigContents): ConfigContents {
-    this.setContents(newContents);
+    if (newContents) {
```

Comment:
If found that running write[Sync] w/o providing `newContents` cause the entire sfdx-project.json file to be overwritten.